astropy/cosmology/units.py

```diff
@@ -1,9 +1,20 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 """Cosmological units and equivalencies."""
 
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Literal
```

Comment:
`Any` and `Literal` are not used at runtime, so I would prefer them to be imported in the `if TYPE_CHEKCING:` block. It would be best if Ruff would manage this automatically. It's not obvious to me why it's not doing that already.

Reply:
> It's not obvious to me why it's not doing that already.

Yeah! I thought we had iterated on this, you and I, and solved it already.

Reply:
Done!